src/plugins/intel_gpu/src/kernel_selector/kernels/rms/rms_kernel_bfyx_opt.cpp

```diff
@@ -0,0 +1,110 @@
+    k.EnableOutputDataType(Datatype::F32);
+    k.EnableInputLayout(DataLayout::bfyx);
+    k.EnableInputLayout(DataLayout::bfzyx);
+    k.EnableOutputLayout(DataLayout::bfyx);
+    k.EnableOutputLayout(DataLayout::bfzyx);
+    k.EnableTensorOffset();
+    k.EnableTensorPitches();
+    k.EnableBatching();
+    k.EnableDifferentTypes();
+    k.EnableDynamicShapesSupport();
+    return k;
+}
+
+JitConstants RMSKernelBfyxOpt::GetJitConstants(const rms_params& params, DispatchData dispatchData) const {
+    auto jit = Parent::GetJitConstants(params, dispatchData);
+
+    if (params.has_dynamic_tensors()) {
+        const auto& input = params.inputs[0];
+        DimensionAccessHelper dims(input);
+        const std::string data_size = toVectorMulString({dims.x(), dims.y(), dims.z()});
+        const std::string lws_0 = "get_local_size(0)";
+        jit.AddConstants({
+            MakeJitConstant("DATA_SIZE", data_size),
+            MakeJitConstant("LWS", lws_0),
+            MakeJitConstant("SLM_SIZE", dispatchData.maxSlmSize)
+        });
+    } else {
+        jit.AddConstants({
+            MakeJitConstant("DATA_SIZE", dispatchData.dataSize),
+            MakeJitConstant("LWS", dispatchData.slmSize),
+            MakeJitConstant("SLM_SIZE", dispatchData.slmSize),
+            MakeJitConstant("LEFTOVERS", dispatchData.leftovers)
+        });
+    }
+    jit.AddConstants({
+        MakeJitConstant("VEC_SIZE", 8),
+        MakeJitConstant("VLOAD", "CAT(vload, VEC_SIZE)"),
+        MakeJitConstant("VSTORE", "CAT(vstore, VEC_SIZE)"),
+        MakeJitConstant("INPUT_VEC_TYPE", "MAKE_VECTOR_TYPE(INPUT0_TYPE, VEC_SIZE)"),
+        MakeJitConstant("ACCUMULATOR_VEC_TYPE", "MAKE_VECTOR_TYPE(ACCUMULATOR_TYPE, VEC_SIZE)"),
+        MakeJitConstant("OUTPUT_VEC_TYPE", "MAKE_VECTOR_TYPE(OUTPUT_TYPE, VEC_SIZE)"),
+        MakeJitConstant("AS_INPUT_VEC_TYPE", "CAT(as_, INPUT_VEC_TYPE)"),
+        MakeJitConstant("AS_ACCUMULATOR_VEC_TYPE", "CAT(as_, ACCUMULATOR_VEC_TYPE)"),
+        MakeJitConstant("TO_ACCUMULATOR_VEC_TYPE", "CAT(convert_, ACCUMULATOR_VEC_TYPE)"),
+        MakeJitConstant("TO_OUTPUT_VEC_TYPE", "CAT(convert_, OUTPUT_VEC_TYPE)"),
+    });
+
+    return jit;
+}
+
+RMSKernelBase::DispatchData RMSKernelBfyxOpt::SetDefault(const rms_params& params) const {
+    DispatchData dispatchData;
+    const auto& input = params.inputs[0];
+
+    auto local_mem_per_wi = 2 * BytesPerElement(params.inputs[0].GetDType());
+    auto max_lws = std::min(params.engineInfo.maxWorkGroupSize, params.engineInfo.maxLocalMemSize / local_mem_per_wi);
+    dispatchData.maxSlmSize = max_lws;
+
+    if (!params.has_dynamic_tensors()) {
+        dispatchData.dataSize = input.X().v * input.Y().v * input.Z().v;
+        dispatchData.dataCount = input.Batch().v * input.Feature().v;
+        dispatchData.slmSize = dispatchData.dataSize / 8;
+        dispatchData.leftovers = dispatchData.dataSize % 8;
+
+        dispatchData.gws[0] = dispatchData.slmSize;
```

Comment:
shouldnt we adjust the WG w.r.t the max # of work items & availalbe  slmSize?

Reply:
As discussed, added code to check whether the data size exceeds the available slm size in `validate` step. At the rms_fusion transformation step, even if the shape of data is static, there is no way to retrieve device info. so it couldn't be implemented at this time.